lib/models/offline_recorder.dart

```diff
@@ -0,0 +1,164 @@
+  }
+
+  int getCurrentMaxStorageSize() {
+    return _maxStorageSize;
+  }
+
+  /// TODO : Only delete some files
+  void _cleanStorage() async {
+    Directory appOfflineNewsFolder = Directory(_appOfflineNewsPath);
+
+    try {
+      var transcriptList = List<String>.empty();
+      appOfflineNewsFolder
+          .list(recursive: false, followLinks: false)
+          .listen((FileSystemEntity file) {
+        debugPrint("[CLEARSTORAGE] path: ${file.path}");
+        if (file.path
+            .substring(file.path.length - "XXXX-XX-XX_transcript.json".length)
+            .startsWith(RegExp(r"[0-9]{4}-[0-9]{2}-[0-9]{2}"))) {
+          transcriptList.add(file.path);
+        }
+      });
+      transcriptList.sort();
+      // Most recent at the end
+      transcriptList = transcriptList.reversed.toList();
+
+      while (await _dirSize(appOfflineNewsFolder) >= (0.75 * _maxStorageSize)) {
+        if (File(transcriptList.last).existsSync()) {
+          await File(transcriptList.last).delete();
+        }
+        transcriptList.removeLast();
+      }
+    } catch (e) {
+      // If an error happen, recreate the whole folder
+      if (await appOfflineNewsFolder.exists()) {
+        appOfflineNewsFolder.delete(recursive: true);
+      }
+
+      await Directory(_appOfflineNewsPath).create(recursive: true);
+    }
+
+    if (await appOfflineNewsFolder.exists()) {
+      appOfflineNewsFolder.delete(recursive: true);
+    }
+
+    await Directory(_appOfflineNewsPath).create(recursive: true);
+  }
+
+  /// Provides the size, in bytes, of a directory
+  Future<int> _dirSize(Directory dir) async {
+    var files = await dir.list(recursive: true).toList();
+    var dirSize = files.fold(0, (int sum, file) => sum + file.statSync().size);
+    return dirSize;
+  }
+
+  Future<void> downloadNews(News news) async {
+    Directory appOfflineNewsFolder = Directory(_appOfflineNewsPath);
+
+    // Missing folder -> creates it, even if it should be created the first time the offline recorder is created
+    if (!(await appOfflineNewsFolder.exists())) {
+      await appOfflineNewsFolder.create(recursive: true);
+    }
+
+    String filePath =
+        _appOfflineNewsPath + "${news.date.substring(0, 10)}_transcript.json";
+    String json = jsonEncode(news);
+
+    // Creates the file
+    final file = File(filePath);
+
+    // Cannot have two transcripts for the same day
+    if (await file.exists()) {
+      await file.delete();
+    }
+
+    await file.writeAsString(json);
+
+    // TODO : Only delete oldest files when overflowing
+    if (await _dirSize(Directory(_appOfflineNewsPath)) >= _maxStorageSize) {
+      _cleanStorage();
+    }
+  }
+
+  /// Load data from local storage. Date must be formated in the following way : <year>-<month>-<day>
+  Future<News> loadNews(String date) async {
+    RegExp regex = RegExp(r"^[0-9]{4}-[0-9]{2}-[0-9]{2}$");
+    if (regex.allMatches(date).isEmpty) {
+      throw UnimplementedError("Date String not well formated");

```

Comment:
```suggestion
      throw FormatException("Date String not well formated");
```

`UnimplementedError` is not adapted to this case, I think `FormatException` suits more.